core/templates/pages/admin-page/activities-tab/admin-dev-mode-activities-tab.component.spec.ts

```diff
@@ -0,0 +1,415 @@
+    it('should not reload a specific exploration if task is running', () => {
+      spyOn(
+        adminTaskManagerService, 'isTaskRunning').and.returnValue(true);
+      expect(component.reloadExploration('expId')).toBeUndefined();
+    });
+
+    it('should not procees if user doesn\'t confirm', () => {
+      mockConfirmResult(false);
+      expect(component.reloadExploration('expId')).toBeUndefined();
+    });
+
+    it('should load explorations', async(() => {
+      const expId = component.demoExplorationIds[0];
+
+      spyOn(adminBackendApiService, 'reloadExplorationAsync')
+        .and.returnValue(Promise.resolve());
+      spyOn(adminTaskManagerService, 'isTaskRunning').and.returnValue(false);
+      spyOn(component.setStatusMessage, 'emit');
+
+      component.reloadExploration(expId);
+      mockConfirmResult(true);
+
+      fixture.whenStable().then(() => {
+        expect(component.setStatusMessage.emit)
+          .toHaveBeenCalledWith('Data reloaded successfully.');
+      });
+    }));
+
+    it('should not load explorations with wrong exploration ID', async(() => {
+      const expId = 'wrong-exp-id';
+
+      spyOn(adminBackendApiService, 'reloadExplorationAsync')
+        .and.returnValue(Promise.reject({
+          data: {
+            error: 'Exploration not found.'
+          }
+        }));
+      spyOn(adminTaskManagerService, 'isTaskRunning').and.returnValue(false);
+      spyOn(component.setStatusMessage, 'emit');
+
+      component.reloadExploration(expId);
+      mockConfirmResult(true);
+
+      fixture.whenStable().then(() => {
+        expect(component.setStatusMessage.emit)
+          .toHaveBeenCalledWith('Server error: Exploration not found.');
+      });
+    }));
+  });
+
+  describe('.printResult', () => {
+    it('should print correct message when numTried' +
+      'is less the number of exploration', () => {
+      const numSucceeded = 0;
+      const numFailed = 0;
+      const numTried = 0;
+
+      spyOn(component.setStatusMessage, 'emit');
+
+      component.printResult(numSucceeded, numFailed, numTried);
+
+      expect(component.setStatusMessage.emit)
+        .toHaveBeenCalledWith('Processing...0/1');
+    });
+
+    it('should print correct message when numTried' +
+      'is not less than the number of exploration', () => {
+      const numSucceeded = 1;
+      const numFailed = 0;
+      const numTried = 1;
+
+      spyOn(component.setStatusMessage, 'emit');
+
+      component.printResult(numSucceeded, numFailed, numTried);
+
+      expect(component.setStatusMessage.emit).toHaveBeenCalledWith(
+        'Reloaded 1 explorations: 1 succeeded, 0 failed.');
+    });
+  });
+
+  describe('.reloadAllExplorations', () => {
+    it('should not reload all exploration if' +
+      'reloading all exploration is possible', () => {
+      component.reloadingAllExplorationPossible = false;
+
+      component.reloadAllExplorations();
+
+      expect(component.reloadAllExplorations()).toBeUndefined();
```

Comment:
Validate whether it's making backendApi call. Ditto elsewhere make sure to assert all the expected behaviors in tests.

Reply:
Done